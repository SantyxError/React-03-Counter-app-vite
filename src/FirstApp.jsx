import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {

  return (
    <>
      <div data-testid="test-title">{title}</div>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  subTitle: "No lo hay, no",
  name: "Santiago Benito",
  title: "No hay titulo",

}