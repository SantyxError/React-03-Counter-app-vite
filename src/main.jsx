import React from 'react'
import ReactDOM from 'react-dom/client'
import { HelloWorldApp } from './HelloWorldApp'
import { FirstApp } from './FirstApp'
import './styles.css'
import { CounterApp } from './CounterApp'



ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <HelloWorldApp /> */}
    <CounterApp value={10} />
  {/*  <FirstApp title='Hola, Soy Goku'></FirstApp> */}
  </React.StrictMode>
)