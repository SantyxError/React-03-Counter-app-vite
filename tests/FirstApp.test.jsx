import { render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"


describe('Pruebas en <FirstApp/>', () => {

  test('debe de hacer match con el snapshot', () => {

    const title = 'Hola, Soy Goku'
    const { container } = render(<FirstApp title={title} />)

    expect(container).toMatchSnapshot()

  })

  test('debe de mostrar el titulo en un h1', () => {

    const title = 'Hola, Soy Goku'
    const { container, getByText } = render(<FirstApp title={title} />)
    expect(getByText(title)).toBeTruthy()

    const h2 = container.querySelector('h2')
    expect(h2.innerHTML).toContain(title);

  })
})