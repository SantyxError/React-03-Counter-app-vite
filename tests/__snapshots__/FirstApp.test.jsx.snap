// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h2>
    Hola, Soy Goku
  </h2>
  <p>
    No lo hay, no
  </p>
  <p>
    Santiago Benito
  </p>
</div>
`;
